frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/InstallNodeAndNpmMojo.java

```diff
@@ -38,8 +38,9 @@ public final class InstallNodeAndNpmMojo extends AbstractFrontendMojo {
 
     /**
      * The version of Node.js to install. IMPORTANT! Most Node.js version names start with 'v', for example 'v0.10.18'
+     * If set to `version_manager`, node version manager will be used to run installation commands
      */
-    @Parameter(property="nodeVersion", required = true)
+    @Parameter(property="nodeVersion", required = false)
```

Comment:
What happens when a node version manager isn't present? We need an assertion somewhere to give users a helpful message about what they need to do

Reply:
i have added assertion for when node version manager usage is enabled but we don't find any, with instructions to install it.

It's relying on nodeVersion value still to install a specific version, I was hoping that your change for reading version files would fit in nicely here, so that potentially we can support more version files than what individual version managers do.

Reply:
Hmmm, what I think we want to do is progressive enhancement, so attempting in this order:

1. Try to use the version manager for detecting the version of Node and installing it, but we do it natively through the command line, we don't try to detect the version of Node ourselves. The reason for this is so that it doesn't lead to unexpected differences between what the user would do themselves and what we're doing.
2. If AND ONLY IF we don't detect a node version manager, we pickup the node version from the file and install it using our own installer. This allows Java-only engineers to still build the project, BUT the version is still controlled from one place
3. Version of Node is declared in the POM and we install it. This is what already exists upstream, we keep this around for backwards compatibility so projects can switch over immediately and progressively move over to the better operating modes **1** and **2**

Along of course with good validation of inputs and informative messages so engineers aren't surprised (e.g. how we're attempting to install node and then everytime we fall back)

Reply:
So how does that play out with installing Yarn and NPM? Not sure yet, need to have a look at the corepack stuff that was recently merged into upstream since that's yet another way to get package managers installed......

Reply:
Yup, sounds good. I'll have a look at this detection now. My initial assumption was incorrect anyway, as some of the node version managers need their files to be around to work across shell sessions

Reply:
Ok, now it lets the version manager to handle the version resolution.